packages/next/next-server/server/next-server.ts

```diff
@@ -220,6 +227,16 @@ export default class Server {
       ),
       flushToDisk: this.nextConfig.experimental.sprFlushToDisk,
     })
+
+    /**
+     * This sets environment variable to be used at the time of SSR by head.tsx.
+     * Using this from process.env allows targetting both serverless and SSR by calling
+     * `process.env.__NEXT_OPTIMIZE_FONTS`.
+     * TODO(prateekbh@): Remove this when experimental.optimizeFonts are being clened up.
+     */
+    if (this.renderOpts.optimizeFonts) {
+      process.env.__NEXT_OPTIMIZE_FONTS = JSON.stringify(true)
```

Comment:
@Timer @timneutkens Does it make sense to automatically add everything in config.experimental to process.env? might help avoid some missed cleanups later on.

Reply:
I was personally not sure about this. Not all experimental flags are required so deep down in the components like this one.